src/foremast/awslambda/api_gateway_event/api_gateway_event.py

```diff
@@ -209,13 +233,15 @@ def create_resource(self, parent_id=""):
             parent_id (str): The resource ID of the parent resource in API Gateway
         """
         resource_name = self.trigger_settings.get('resource', '')
-        resource_name = resource_name.replace("/", "")
-        created_resource = self.client.create_resource(restApiId=self.api_id,
-                                                       parentId=parent_id,
-                                                       pathPart=resource_name)
-        resource_id = created_resource['id']
-        self.log.info("Successfully created resource")
-        return resource_id
+        resource_name = resource_name.replace('/', '')
+        if not self.resource_id:
+            created_resource = self.client.create_resource(restApiId=self.api_id,
+                                                           parentId=parent_id,
+                                                           pathPart=resource_name)
+            self.resource_id = created_resource['id']
```

Comment:
I don't like to do this here but its worth moving create_resource call out of find_resource_id function.

When we initialize on a fresh deployment self.resource_id will be None but it is needed to pass to attach_method. Would moving the attach method call from the end of this file into this create_resource function be less bad?